mypy/checkexpr.py

```diff
@@ -330,21 +337,42 @@ def try_infer_partial_type(self, e: CallExpr) -> None:
                                                                        list(full_item_types))
                                 del partial_types[var]
 
+    def apply_function_plugin(self,
+                              arg_types: List[Type],
+                              inferred_ret_type: Type,
+                              arg_kinds: List[int],
+                              formal_to_actual: List[List[int]],
+                              args: List[Expression],
+                              num_formals: int,
+                              fullname: Optional[str]) -> Type:
+        """Use special case logic to infer the return type for of a particular named function.
+
+        Return the inferred return type.
+        """
+        formal_arg_types = [None] * num_formals  # type: List[Optional[Type]]
+        for formal, actuals in enumerate(formal_to_actual):
+            for actual in actuals:
+                formal_arg_types[formal] = arg_types[actual]
```

Comment:
Aren't there some edge cases where map_actuals_to_formals() returns overlapping mappings? (IIRC related to `*args` and worse.)